### ISSUE TITLE
An Error Mentioned by Bug Report, I have no idea what it is just said to post it as it might help others

### ISSUE THREAD

**Issue by AnkithVijay** [flutter/flutter-intellij#3443]
## What happened

An Error Mentioned by Bug Report, I have no idea what it is just said to post it as it might help others

## Version information

Android Studio `3.4` • Flutter plugin `io.flutter 35.1.1` • Dart plugin `183.6270`

Error getting Flutter sdk information.

## Exception

null

```
java.lang.IndexOutOfBoundsException: Wrong offset: 3018. Should be in range: [0, 2970]
	at com.intellij.openapi.editor.impl.LineSet.findLineIndex(LineSet.java:172)
	at com.intellij.openapi.editor.impl.DocumentImpl.getLineNumber(DocumentImpl.java:1008)
	at io.flutter.editor.FliteredIndentsHighlightingPass.getGuideLineRange(FliteredIndentsHighlightingPass.java:202)
	at io.flutter.editor.FliteredIndentsHighlightingPass.onWidgetIndentsChanged(FliteredIndentsHighlightingPass.java:252)
	at io.flutter.editor.WidgetIndentsHighlightingPass.updateHitTester(WidgetIndentsHighlightingPass.java:559)
	at io.flutter.editor.WidgetIndentsHighlightingPass.setOutline(WidgetIndentsHighlightingPass.java:547)
	at io.flutter.editor.WidgetIndentsHighlightingPass.run(WidgetIndentsHighlightingPass.java:522)
	at io.flutter.editor.WidgetIndentsHighlightingPassFactory.runWidgetIndentsPass(WidgetIndentsHighlightingPassFactory.java:299)
	at io.flutter.editor.WidgetIndentsHighlightingPassFactory.lambda$null$1(WidgetIndentsHighlightingPassFactory.java:196)
	at com.intellij.openapi.application.TransactionGuardImpl$2.run(TransactionGuardImpl.java:315)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.doRun(LaterInvocator.java:435)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.runNextEvent(LaterInvocator.java:419)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:403)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:762)
	at java.awt.EventQueue.access$500(EventQueue.java:98)
	at java.awt.EventQueue$3.run(EventQueue.java:715)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:732)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:755)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:704)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:391)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

**Comment by devoncarew** [comment]
Thanks for the feedback!

@jacob314, this looks like an IndexOutOfBoundsException in the new UI Guides code.

**Comment by jacob314** [comment]
Thank you for reporting this issue. I believe I have a fix at https://github.com/flutter/flutter-intellij/pull/3445
I'm now defensive about handling range markers that are off the end of the document absolutely everywhere we call the getLineNumber method.

**Comment by jacob314** [comment]
The fix has been published as a version 35.2of the plugin. Please let me know if you see any other issues using the UI guides.
